### ISSUE TITLE
Android shield + handler

### ISSUE THREAD

**Issue by ncoen97** [gbumps/react-native-screenguard#58]
In the provided android video example, I can see the screenshot gets "shielded" by  covering the screen with green, but in my case (using 0.3.6) when using the register function the screenshot gets blocked entirely.

I think that's causing the android handler not to trigger, and I need both features working at the same time (blocking + handling).

Is there a way to achieve this? Or is there a mismatch between the code and the documentation?

Thanks.

**Comment by gbumps** [comment]
Hello @ncoen97, version 1.0.0 is successfully released. Please check document, install and rebuild.

Thanks!

**Comment by ncoen97** [comment]
Hello @gbumps.

I've tested with the las version (android 14) and the screenshot handler is still getting blocked.

I've followed the "Post-installation setting for Android" steps and this is my current implementation

```javascript
  useFocusEffect(
    useCallback(() => {
      ScreenGuardModule.register({backgroundColor: '#252C30'});
      ScreenGuardModule.registerScreenshotEventListener(
        false,
        openModal,
      );
      ScreenGuardModule.registerScreenRecordingEventListener(
        openModal,
      );
      return ScreenGuardModule.unregister;
    }, [openModal]),
  );

```

After looking at `ScreenGuardModule.java`, I noticed the `activateShield` function is still setting the `FLAG_SECURE` flag.

I tried to make it work by removing the `setFlags` and running my app via `react-native run-android`. It did let the screenshot go through, but it's not showing the screen blocker while taking the screenshot.

**Comment by gbumps** [comment]
As told in previous comment, handler won't work until the `FLAG_SECURE` is removed. This is a characteristic of the Android platform, and there is no way to handle this issue. So basically you cannot have both handler + screen block at the same time on Android.

You can somehow get the info of the latest image (usually saved in Photos or in ScreenShot folder) and check if it is a newly screenshot or not. But it's not always reliable since you may have to ask permission Storage before proceed, and users can customize file saving on other location.

**Comment by gbumps** [comment]
As told in previous comment, while the FLAG_SECURE is turned on, all screenshot and screen record attempting is fully blocked by Android system. This is a characteristic of the platform, and there is no way to handle this issue. So basically you cannot have both handler + screen block at the same time on Android.

Handler is still working in case you use another screenshot or screen recording app, or screenshot system app (scroll down from top to see Notification and Panel setting -> screenshot) by device manufacturer (but may vary).

There is another way, You can somehow get the info of the latest image (usually saved in Photos or in ScreenShot folder) and check if it is a newly screenshot or not. But it's not always reliable since you may have to ask permission Storage before proceed, and users can customize file saving on other locations.

**Comment by ncoen97** [comment]
I don't get what's the difference between `register` and `registerWithoutEffect` in android then. Both are blocking the screenshot entirely and I can't achieve the functionality seen in the android example video (screenshot gets covered by the black color and handler gets executed).

Is that correct? or am I missing something?

Thanks for the follow up.

**Comment by gbumps** [comment]
`register` is cover screen with a background color while `registerWithoutEffect` is just screen guarding only without background color or any other effects like blur view or image view.

**Comment by ncoen97** [comment]
Okay.

The `register` function is not working for me, and as per my last comment, the `activateShield` function is still setting the `FLAG_SECURE` flag and NOT showing the background color nor executing the handler as expected.

**Comment by gbumps** [comment]
if you want to receive handler, you must use `registerScreenshotEventListener` first. You can check for my example [here](https://github.com/gbumps/react-native-screenguard/blob/master/example/App.tsx).

The background color you received on Android is after calling any `register` function and go out of your application for a while, and the jump right back in, you will see a background color as you've set up. After 2 or 3 secs based on your settings, it's will automatically come back to your main application. 

```
the activateShield function is still setting the FLAG_SECURE flag..... 
```

Still don't understand your intention here ?

**Comment by ncoen97** [comment]
Is this code from the example working for you in android?

```javascript
<Pressable
  onPress={() => {
    ScreenGuardModule.register({
      backgroundColor: color,
      timeAfterResume: 2000,
    });
    ScreenGuardModule.registerScreenshotEventListener(false, event =>
      Alert.alert(`register screenshot,`),
    );
    ScreenGuardModule.registerScreenRecordingEventListener(_ =>
      Alert.alert('register screen record'),
    );
    setCurrentState('1');
  }}>
  <Text
    style={{
      color: currentState === '1' ? '#00FF00' : Colors.white,
    }}>
    Turn on screenguard with color
  </Text>
</Pressable>
``` 

I want to replicate the same behaviour it has on iOS but also in android, and I'm not able to do that.

I need the screenshot to get blocked in some way and the handler to get executed. How can I achieve that?

**Comment by gbumps** [comment]
I've already stated on previous comment that you can't do it on Android cause it a characteristic of the platform, when the flag FLAG_SECURE is active, it means all of screen capture and screen recording is blocked by system, so that it's why the handler is not get executed except when you use a screenshot or screen recording application, handler still active and got executed normally, but may vary on some devices.

on iOS you can capture and image and got handler executed because it's just only the filter itself and the system it's not blocking anything.

**Comment by gbumps** [comment]
I think I might put an end to this and not let it go too far, by a short summarization:

on Android, screenshot handler will not executed when the FLAG_SECURE is turned on. However, when using a screenshot application (like AZ, Xrecoder, etc...), handler will be triggered normally, but may vary on some devices.

The background color, or the blur image view you saw on the example screen recording video is one of my custom implementation on Android, and of course, it's way different from the iOS and I've stated very clearly in README and TS code definition. Please revise carefully before submitting new issue.
